contracts/smart-wallet/dynamic/DynamicAccount.sol

```diff
@@ -41,14 +46,10 @@ contract DynamicAccount is Account, BaseRouter {
                             Public Overrides
     //////////////////////////////////////////////////////////////*/
 
-    /// @dev See {IERC165-supportsInterface}.
-    function supportsInterface(bytes4 interfaceId) public view virtual override(Account, BaseRouter) returns (bool) {
-        return
-            interfaceId == type(IBaseRouter).interfaceId ||
-            interfaceId == type(IRouter).interfaceId ||
-            interfaceId == type(IERC1155Receiver).interfaceId ||
-            interfaceId == type(IERC721Receiver).interfaceId ||
-            super.supportsInterface(interfaceId);
```

Comment:
Don't we need this?

Reply:
Yes we do, but this goes into the `AccountExtension` smart contract so that this function can be upgraded in the future.